src/check_release.rs

```diff
@@ -325,16 +338,48 @@ pub(super) fn run_check_release(
             print_failed_lint(config, semver_query, results)?;
         }
 
-        let required_bump = if required_versions.contains(&RequiredSemverUpdate::Major) {
-            RequiredSemverUpdate::Major
-        } else if required_versions.contains(&RequiredSemverUpdate::Minor) {
-            RequiredSemverUpdate::Minor
-        } else {
-            unreachable!("{:?}", required_versions)
-        };
+        for (semver_query, results) in results_with_warnings {
+            suggested_versions.push(overrides.effective_required_update(semver_query));
+            config.log_info(|config| {
+                writeln!(
+                    config.stdout(),
+                    "\n--- warning: {}: {} ---\n",
+                    semver_query.id,
+                    semver_query.human_readable_name
+                )?;
+                Ok(())
+            })?;
 
-        config
-            .shell_print(
+            print_failed_lint(config, semver_query, results)?;
+        }
+
+        let required_bump = required_versions.iter().max().copied();
+        let suggested_bump = suggested_versions.iter().max().copied();
+
+        if let Some(suggested_bump) = suggested_bump {
+            // only print warning version if it is greater than the required version bump
+            if required_bump.map_or(true, |required_bump| required_bump < suggested_bump) {
+                config.shell_warn(format_args!(
+                    "warning checks suggest new {} version",
+                    suggested_bump.as_str()
+                ))?;
+            }
+
+            config.shell_warn(format_args!(
+                "{} major and {} minor warning checks failed",
+                suggested_versions
+                    .iter()
+                    .filter(|x| *x == &RequiredSemverUpdate::Major)
+                    .count(),
+                suggested_versions
+                    .iter()
+                    .filter(|x| *x == &RequiredSemverUpdate::Minor)
+                    .count(),
+            ))?;
+        }
+
+        if let Some(required_bump) = required_bump {
```

Comment:
It's strange that by this point, we've already printed out summary information about warnings, and now we print a line titled "Summary" with more information.